drivers/mipi_dbi/mipi_dbi_nxp_dcnano_lcdif.c

```diff
@@ -0,0 +1,430 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_dcnano_lcdif_dbi
+
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/mipi_dbi.h>
+#include <zephyr/dt-bindings/mipi_dbi/mipi_dbi.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <fsl_lcdif.h>
+#include <fsl_mipi_dsi.h>
+
+LOG_MODULE_REGISTER(mipi_dbi_nxp_dcnano_lcdif, CONFIG_DISPLAY_LOG_LEVEL);
+
+#if CONFIG_MIPI_DSI
+#define MIPI_DSI_MAX_PAYLOAD_SIZE 0xFFFF
+#endif
+
+struct mcux_dcnano_lcdif_dbi_data {
+	struct k_sem transfer_done;
+	uint8_t *data;
+	uint8_t dsiFormat;
+	uint16_t pitch;
+	uint16_t heightEachWrite;
+	uint16_t heightSent;
+	uint16_t height;
+	uint16_t width;
+	uint16_t stride;
+};
+
+struct mcux_dcnano_lcdif_dbi_config {
+	LCDIF_Type *base;
+	MIPI_DSI_HOST_Type *dsi_base;
+	void (*irq_config_func)(const struct device *dev);
+	lcdif_dbi_config_t dbi_config;
+	lcdif_panel_config_t panel_config;
+	const struct pinctrl_dev_config *pincfg;
+	const struct gpio_dt_spec reset;
+};
+
+static void mcux_dcnano_lcdif_dbi_isr(const struct device *dev)
+{
+	const struct mcux_dcnano_lcdif_dbi_config *config = dev->config;
+	struct mcux_dcnano_lcdif_dbi_data *lcdif_data = dev->data;
+	uint32_t status;
+
+	status = LCDIF_GetAndClearInterruptPendingFlags(config->base);
+
+	if (0 != (status & kLCDIF_Display0FrameDoneInterrupt))
+	{
+		/* Handle multi chunk with mipi. */
+#if CONFIG_MIPI_DSI
```

Comment:
I'm unclear why we have dependencies on `CONFIG_MIPI_DSI` here?

From what I understand about the DC8000 peripheral, it can support MIPI DBI (8080 and 6800), and it can support MIPI DPI, which would be sent via a DSI phy to the display.

A few questions here:
- The MIPI DPI support would be used for displays that operate in video mode (IE no graphics RAM, require constant refresh). An example would be the RK055HDMIPI4MA0 display
- The DSI PHY on this SOC supports sending MIPI DSI commands directly, right? This would be used for displays like the G1120B0MIPI, correct? Or can the DC8000 still drive those displays?

If I'm understanding the peripheral right, then we probably need something like the following:
- A new compatible for the peripheral- this is clearly different from the DCNANO IP present on the RT500
- A MIPI-DBI driver that supports driving displays in 6800 or 8080 mode
- A MIPI-DSI driver for the MIPI-DPHY (if this IP is reused from the RT500, we can reuse the existing driver)
- Additions to the DCNANO LCDIF driver to support this new DC8000 IP (using the trick I mentioned for multiple compatibles)

Critically, the DC8000 should only have *one* compatible string. Users can select which driver gets used with the IP based on which Kconfigs they enable. We get into a bit of a strange situation though, since it seems this IP can be used in MIPI-DBI mode, or as a display driver (enabled with `CONFIG_DISPLAY`). We probably need to make it so that `CONFIG_DISPLAY_MCUX_DCNANO_LCDIF` depends on `CONFIG_MIPI_DBI` being disabled.

Reply:
_The DSI PHY on this SOC supports sending MIPI DSI commands directly, right? This would be used for displays like the G1120B0MIPI, correct? Or can the DC8000 still drive those displays?_
Yes, on RT700 the panel G1120B0MIPI is driven by DC8000 and MIPI DSI combined. We can also use the RT500 legacy way to drive G1120B0MIPI by sending MIPI DSI commands and data directly on APB, but since RT700 does not have smartdma, this way can be rather slow, and does not support pixel format convertion, eg. the frame buffer in memory is stored in RGB888 format while the data sent on bus in D16RGB565 format as G1120B0MIPI required. To use both DC8000 and the MIPI DSI to drive the panel, there are many additional actions to do on the MIPI DSI side. Since the internal interface between these 2 modules have limited buffering, if the data to be sent is too large, the driver will have to chop up the data and sent one piece at a time. So for G1120B0MIPI the CONFIG_MIPI_DSI has to be enabled, for displays in 6800 or 8080 mode, there is no need to enable CONFIG_MIPI_DSI.
_A new compatible for the peripheral- this is clearly different from the DCNANO IP present on the RT500_
Yes I created a new compatible nxp,dcnano_lcdif_dbi for this driver drivers/mipi_dbi/mipi_dbi_nxp_dcnano_lcdif.c, take reference from the compatible nxp,dcnano_lcdif of drivers/display/display_mcux_dcnano_lcdif.c(not sure if it is a good practice since they belong to different driver groups). They all use the same IP DC8000 on RT700, but as you pointed out DC8000 can either be in DBI or in DPI mode so only one compatible at the same time, so in my test example, I change the compatible in the display's overlay(still not sure if it is a good practice since I cannot find similar cases).
https://github.com/zephyrproject-rtos/zephyr/commit/476b7bb1df6786620fd759ce2b6e9ab25703cc23#diff-1b9a0139b8cc287d78c4bb150fc04ccc161c82f8fea192fe9aa1ecc9a177c18aR18
_We probably need to make it so that CONFIG_DISPLAY_MCUX_DCNANO_LCDIF depends on CONFIG_MIPI_DBI being disabled._
Thanks, will update the dependency.